src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -43,15 +53,12 @@ static ElasticApmModule()
 		private Transaction _currentTransaction;
 
 		private HttpApplication _httpApp;
-
-		private static Version AspNetVersion => typeof(HttpRuntime).Assembly.GetName().Version;
-		private static string ClrDescription => PlatformDetection.FrameworkDescription;
 		private static Version IisVersion => HttpRuntime.IISVersion;
 
-		private static void SetServiceInformation(Service service)
+		private static void SetServiceInformation(Service service, string aspNetVersion)
 		{
-			service.Framework = new Framework { Name = "ASP.NET", Version = AspNetVersion.ToString() };
-			service.Language = new Language { Name = "C#" }; //TODO
+			service.Framework = new Framework { Name = "ASP.NET", Version = aspNetVersion };
+			service.Language = new Language { Name = "C#" };
```

Comment:
You removed the `TODO` :) This can be VB or F#. We have https://github.com/elastic/apm-agent-dotnet/issues/196. I think we don't have a common way to handle these, but I like "honest" code and usually just put a `TODO` there in case something is to be improved.

Reply:
NP - Reverted.